multi-channel-chat/nats-channel/src/lib.rs

```diff
@@ -0,0 +1,123 @@
+use chatlog::*;
+use outbound::*;
+use serde::{Deserialize, Serialize};
+use wasmbus_rpc::actor::prelude::*;
+use wasmcloud_interface_messaging::*;
+use wasmcloud_interface_numbergen::*;
+
+#[allow(dead_code)]
+mod chatlog;
+
+#[allow(dead_code)]
+mod outbound;
+
+const CHANNEL_NAME: &str = "nats";
+const API_ACTOR: &str = "mcchat/api";
+const RTCHAT_TOPIC: &str = "rtchat.messages";
+
+#[derive(Debug, Default, Actor, HealthResponder)]
+#[services(Actor, MessageSubscriber, Outbound)]
+struct NatsChannelActor {}
+
+#[async_trait]
+impl Outbound for NatsChannelActor {
+    async fn publish_message(&self, ctx: &Context, arg: &OutboundMessage) -> RpcResult<bool> {
+        let pubber = MessagingSender::new();
+        let res = pubber
+            .publish(
+                ctx,
+                &PubMessage {
+                    subject: RTCHAT_TOPIC.to_string(),
+                    reply_to: None,
+                    body: serde_json::to_vec(arg).unwrap(),
+                },
+            )
+            .await;
+        Ok(res.is_ok())
+    }
+}
+
+/// Implementation of Chat Log actor trait methods
+#[async_trait]
+impl MessageSubscriber for NatsChannelActor {
+    async fn handle_message(&self, ctx: &Context, msg: &SubMessage) -> RpcResult<()> {
+        let reply_topic = msg
+            .reply_to
+            .as_ref()
+            .unwrap_or(&"dedletter".to_string())
+            .to_string();
+        match serde_json::from_slice::<IncomingMessage>(&msg.body) {
+            Ok(im) => {
+                let logger = ChatlogSender::to_actor(API_ACTOR);
+                let numgen = NumberGenSender::new();
+                let guid = numgen.generate_guid(ctx).await.unwrap_or("n/a".to_string());
+                match logger
+                    .write_message(
+                        ctx,
+                        &CanonicalChatMessage {
+                            body: im.body,
+                            channel_name: CHANNEL_NAME.to_string(),
+                            id: guid,
+                            source_user: im.user_name,
+                        },
+                    )
+                    .await
+                {
+                    Ok(resp) => {
+                        let ack = Ack {
+                            accepted: resp.accepted,
+                            reason: resp.reason,
+                        };
+                        let _ = pub_ack(ctx, &reply_topic, ack).await;
+                    }
+                    Err(e) => {
+                        let _ = pub_fail(ctx, &reply_topic, format!("{}", e)).await;
+                    }
+                }
+            }
+            Err(_) => {
+                let _ = pub_fail(ctx, &reply_topic, "Bad Request".to_string()).await;
+            }
+        }
+        Ok(())
+    }
+}
+
+async fn pub_ack(ctx: &Context, topic: &str, ack: Ack) {
+    let publisher = MessagingSender::new();
+    let _ = publisher
```

Comment:
This let is unnecessary and will read weird to non-rust people. Should just be able to drop it

Reply:
this is done so that we can invoke it and we don't care about the result.

Reply:
I don't know an alternate syntax for "do a thing and we don't care about the result" without this, otherwise you get the "must use a result" error